tests/WidgetTest.php

```diff
@@ -117,13 +150,25 @@ public function testInjection(): void
         $this->assertInstanceOf(Injectable::class, $widget->getInjectable());
     }
 
-    public function testFactoryExceptionWidget(): void
+    public function testWidgetThrownExceptionForNotInitializeWidgetFactory(): void
     {
-        $reflection = new ReflectionClass(WidgetFactory::class);
-
-        $widgetFactory = $reflection->newInstanceWithoutConstructor();
-        $this->setInaccessibleProperty($widgetFactory, 'factory', null);
-        $this->expectException(RuntimeException::class);
+        $widgetFactoryReflection = new ReflectionClass(WidgetFactory::class);
+        $reflection = new ReflectionClass($widgetFactoryReflection->newInstanceWithoutConstructor());
+        $property = $reflection->getProperty('factory');
+        $property->setAccessible(true);
+        $property->setValue($widgetFactoryReflection, null);
+        $property->setAccessible(false);
+
+        $this->expectException(WidgetFactoryInitializationException::class);
+        $this->expectExceptionMessage('Widget factory should be initialized with WidgetFactory::initialize() call.');
         TestWidget::widget()->id('w0')->render();
     }
+
+    public function testWidgetFactoryInitializationExceptionMessages(): void
```

Comment:
Maybe merge this test with `testWidgetThrownExceptionForNotInitializeWidgetFactory` ?

Reply:
In what way? In one case, the exception is thrown, and in the other, the custom methods of the exception class are checked.

Reply:
Ok